subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/AbiExtractingClasspathResourceHasher.java

```diff
@@ -37,10 +35,9 @@
 public class AbiExtractingClasspathResourceHasher implements ResourceHasher {
     private static final Logger LOGGER = LoggerFactory.getLogger(AbiExtractingClasspathResourceHasher.class);
 
-    private HashCode hashClassBytes(InputStream inputStream) throws IOException {
+    private HashCode hashClassBytes(byte[] classBytes) {
         // Use the ABI as the hash
-        byte[] classBytes = ByteStreams.toByteArray(inputStream);
```

Comment:
instead of reading into multiple buffers (starting at 8k) and copying back to the final buffer at the end, we can take advantage of the size property of the file and zipfile